docs/debuginfod.md

```diff
@@ -0,0 +1,45 @@
+# debuginfod client
+
+In order to symbolize ingested profiles, Parca needs to have debug information
+for the binaries that are being profiled. Debug information
+typically comprises ELF object files, DWARF debug data, and source code
+required for symbolization. However, sometimes, application packages distributed
+by various Linux distros strip away debug information to minimize the size of
+the binaries. Thankfully, there are public vendor distributed servers which make distro
+specific debug info files available to the community.
+
+debuginfod is an HTTP file server that serves debug information to clients
+based on the build IDs of the binaries. You can find out the build ID of a
+binary using the `file` command on Linux. Here is an example to find out the
+build ID of a zsh shell:
+
+```
+$ file /bin/zsh
+
+/bin/zsh: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=24fcd0179bb3aa797de6a570c2359e528f7638c0, for GNU/Linux 3.2.0, stripped
+```
+Parca integrates with debuginfod to query for upstream debuginfod files and then
+stores them for potential later use. The default debuginfod server used by Parca is
+at https://debuginfod.elfutils.org .
+
+# Implementation
+
+Primarily, Parca looks for the relevant debug information files in its default
+symbol store. However, if debug info files are not found in the symbol store,
+Parca will try to fetch corresponding debuginfo files from the upstream
+debuginfod servers, and store them in the Parca symbol store, associated with
+the unique build id of the object files.
```

Comment:
we use "build id" and "build ID" inconsistently in places, I don't really mind one or the other but it should be consistent